Charon/filetypes/GCodeFile.py

```diff
@@ -68,9 +69,13 @@ def parseHeader(stream: IO[bytes], *, prefix: str = "") -> Dict[str, Any]:
 
             flavor = metadata.get("flavor", None)
             if flavor == "Griffin":
-                if metadata["header_version"] != "0.1":
+                if not GCodeFile.SkipHeaderValidation and metadata["header_version"] != "0.1":
                     raise InvalidHeaderException("Unsupported Griffin header version: {0}".format(metadata["header_version"]))
-                GCodeFile.__validateGriffinHeader(metadata)
+                try:
+                    GCodeFile.__validateGriffinHeader(metadata)
+                except InvalidHeaderException as ex:
+                    if not GCodeFile.SkipHeaderValidation:
+                        raise ex
```

Comment:
Why not a simple
```py
if not GCodeFile.SkipHeaderValidation:
    GCodeFile.__validateGriffinHeader(metadata)
```
? Isn't that more straightforward than the whole try..except? Or maybe I'm missing something?

Reply:
We do this because apparently, the validate function also parses part of the header.

Rewriting it so that it first validates, then parses would be a bit out of the scope of this small change.

This is also why we moved the parts in the validation around.

Reply:
Yes, that's what I meant when I wrote that :-)

... now that I'm re-checking the function, I can see what you say. Strange, I could've sworn that something was saved in that function. I remember trying it without. -- Perhaps I did something wrong when I originally tried that.

If we're not _both_ wrong, we could indeed try to resolve it the way you originally suggested here.